### ISSUE TITLE
3.12.3 release binaries do not include commits present in 3.12.3 Git tag

### ISSUE THREAD

**Issue by kevinburkemeter** [protocolbuffers/protobuf#7632]
The commit before the v3.12.3 tagged release is 28cc693f3d05bab5f779b6fc8d514868130e81c1. I would expect to see this commit present in releases tagged for v3.12.3. Unfortunately, it appears as if it is not present; if I download the v3.12.3 release from the Github releases page, and run it, I do not see that "Generated protocol buffer code" message in the generated Python code.

When I install protoc from Homebrew, it checks out the Git tag and compiles, so I do see the message. This is a problem because some of my team members use Linux and some of them use Mac. This means that the generated protobuf file thrashes back and forth when different members of the team generate it.

I can't inspect the binary but I did run strings on it and I did not see the message in the compiled binary. Compare:

```
~/src/github.com/meterup/proto/protoc-3.12.3-linux-x86_64
± (python*) $ strings ./bin/protoc| grep 'Generated protocol'

~/src/github.com/meterup/proto/protoc-3.12.3-linux-x86_64
± (python*) $ strings /usr/local/Cellar/protobuf/3.12.3/lib/libprotoc.23.dylib | grep 'Generated protocol'
"""Generated protocol buffer code."""
```

Could you consider creating a new release that contains exactly the commits described in the release tag? This is the best way to ensure that things are predictable. 

It seems like others have previously reported problems with the Protobuf release process so this may be worth investing some time in. Alternatively, include metadata about the Git commit in the `.zip` file so that it can be compared against the actual Git tag corresponding to the release.

https://github.com/protocolbuffers/protobuf/issues/7560 
https://github.com/protocolbuffers/protobuf/issues/7600
https://github.com/protocolbuffers/protobuf/issues/7627

**Comment by haberman** [comment]
I think I may have accidentally tagged v3.12.3 from the master branch instead of the release branch.

I agree that the release tooling should tag the release automatically so that it is not possible to make this mistake.

I also agree that I should cut a v3.12.4 release to fix this.

**Comment by kevinburkemeter** [comment]
Thanks!

**Comment by jschueller** [comment]
when is the 3.12.4 due ?

**Comment by xkisu** [comment]
I'm also experiencing this due to https://github.com/protocolbuffers/protobuf/pull/7496, C++ files generated by 3.12.3 don't contain the typo fix, while the Protobuf C++ 3.12.3 sources and headers fetched by vcpkg do. When will there be a fix for this?

**Comment by xkisu** [comment]
@haberman This has been an issue for well over a month now - is there an eta on a fix?

**Comment by kevinburkemeter** [comment]
See also https://github.com/protocolbuffers/protobuf/issues/7637, https://github.com/Homebrew/homebrew-core/pull/58104

**Comment by acozzette** [comment]
I released 3.12.4 this week, which fixes the issue with 3.12.3.
